actionpack/lib/action_controller/metal/url_for.rb

```diff
@@ -4,7 +4,10 @@ module ActionController
   #
   # In addition to <tt>AbstractController::UrlFor</tt>, this module accesses the HTTP layer to define
   # url options like the +host+. In order to do so, this module requires the host class
-  # to implement +env+ and +request+, which need to be a Rack-compatible.
+  # to implement +env+ which need to be Rack-compatible and +request+
+  # which is either instance of +ActionDispatch::Request+ or respond
```

Comment:
@arthurnn 
`or an object that responds to... `?

Thanks @shunsukeaida Fixing it.